test/parallel/test-events-on-async-iterator.js

```diff
@@ -372,6 +373,28 @@ async function abortableOnAfterDone() {
   });
 }
 
+async function abortListenerRemovedAfterComplete() {
```

Comment:
might be worth adding a test ensuring the listener is removed after `abort` as well